security-csrf/src/test/java/io/micronaut/security/csrf/filter/CsrfFilterConfigurationTest.java

```diff
@@ -0,0 +1,66 @@
+package io.micronaut.security.csrf.filter;
+
+import io.micronaut.context.annotation.Property;
+import io.micronaut.core.order.OrderUtil;
+import io.micronaut.core.order.Ordered;
+import io.micronaut.core.util.PathMatcher;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.MediaType;
+import io.micronaut.security.filters.SecurityFilter;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Property(name = "micronaut.security.csrf.filter.regex-pattern", value = "^(?!\\/login).*$")
+@MicronautTest(startApplication = false)
+class CsrfFilterConfigurationTest {
+
+    @Inject
+    CsrfFilterConfiguration csrfFilterConfiguration;
+
+    @Inject
+    CsrfFilter csrfFilter;
+
+    @Inject
+    SecurityFilter securityFilter;
+
+    @Test
+    void orderOfFilters() {
+        List<Ordered> filters = new ArrayList<>(List.of(csrfFilter, securityFilter));
+        OrderUtil.sort(filters);
+        assertInstanceOf(SecurityFilter.class, filters.get(0));
+
+        filters = new ArrayList<>(List.of(securityFilter, csrfFilter));
+        OrderUtil.sort(filters);
+        assertInstanceOf(SecurityFilter.class, filters.get(0));
+    }
+
+    @Test
+    void defaultMethods() {
+        assertEquals(Set.of(HttpMethod.POST, HttpMethod.DELETE, HttpMethod.PUT, HttpMethod.PATCH), csrfFilterConfiguration.getMethods());
+    }
+
+    @Test
+    void defaultContentType() {
+        assertEquals(Set.of(MediaType.APPLICATION_FORM_URLENCODED_TYPE, MediaType.MULTIPART_FORM_DATA_TYPE), csrfFilterConfiguration.getContentTypes());
+    }
+
+    @Test
+    void regexPatternCanBeChanged() {
+        String regexPattern = csrfFilterConfiguration.getRegexPattern();
+        assertFalse(PathMatcher.REGEX.matches(csrfFilterConfiguration.getRegexPattern(), "/login"));
+        assertTrue(PathMatcher.REGEX.matches(csrfFilterConfiguration.getRegexPattern(), "/todo/list"));
+        assertEquals("^(?!\\/login).*$", regexPattern);
```

Comment:
We need to check whether this approach is vulnerable to path traversal attacks. e.g. if the attacker triggers a request to `/login/../todo/list`

Reply:
that it is regular expression used to match the request's path the server filter applies. 

Not sure how an attacker can attack if they sent a request to `/login/.../todo/list` they will get a 404

Reply:
I'm just not super confident in the security of using filter patterns for matching what is essentially routes. If the router does some shenanigans with the URI, such as "simplifying" `/login/../todo/list` to `/todo/list`, then this form of exclusion using regex may become vulnerable.

I'd be more comfortable if we could match this regex against the URI of the resolved route, because that is not attacker-controlled. Would that be possible?

Reply:
what API exposes the `URI of the resolved route`? 
> I'm just not super confident in the security of using filter patterns for matching what is essentially routes. If the router does some shenanigans with the URI, such as "simplifying" /login/../todo/list to /todo/list, then this form of exclusion using regex may become vulnerable.

If the router redirects a request with path `/login/../todo/list` to `/todo/list` then I think we have a vulnerability in the router. 

However, I am ok doing  an extra check in the Csrf filter. However, What API exposes the `URI of the resolved route`? I looked into `RouteMatch` but I don't see any path there.

Reply:
UriRouteMatch has a getUri method which is normalized to some extent by the router (leading slash and trailing parameters removed). Maybe that could work

Reply:
something like https://github.com/micronaut-projects/micronaut-security/pull/1826/commits/13c9cfe7898bb0d246da96be36bf2e597fbd577c ?